pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -0,0 +1,88 @@
+package historian
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/annotations"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/state"
+)
+
+// AnnotationStateHistorian is an implementation of state.Historian that uses Grafana Annotations as the backing datastore.
+type AnnotationStateHistorian struct {
```

Comment:
I'm keeping in mind that the state history may not necessarily _replace_ annotations, this isn't really an explored problem yet.

Even if we keep annotations, we can still treat them as an implementation of this interface, because it's still effectively recording history - we can easily keep them around with a composite implementation.

This approach here does not mean we are getting rid of annotations.